src/main/java/bio/terra/service/dataset/DatasetSchemaUpdateValidator.java

```diff
@@ -51,7 +52,7 @@ private void validateDatasetSchemaUpdate(DatasetSchemaUpdateModel updateModel, E
         Object[] requiredColumns =
             updateModel.getChanges().getAddColumns().stream()
                 .flatMap(c -> c.getColumns().stream())
-                .filter(c -> c.isRequired())
+                .filter(c -> Objects.requireNonNullElse(c.isRequired(), false))
```

Comment:
Good catch, I would have guessed that null values would be excluded in the filter!